ScoutSuite/frontend/src/partials/aws/ec2.regions.id.vpcs.id.security_groups.id/RulesList/index.js

```diff
@@ -0,0 +1,88 @@
+import React, { useContext }  from 'react';
+import get from 'lodash/get';
+
+import { PartialContext, PartialPathContext } from '../../../../components/Partial/context';
+import { PartialValue } from '../../../../components/Partial';
+
+import './style.scss';
+
+
+const RulesList = () => {
+  const ctx = useContext(PartialContext);
+  const basePath = useContext(PartialPathContext);
+  const value = get(ctx.item, basePath);
+
+  return (
+    <ul className="rules-list">
+      {Object.entries(value.protocols).map(([name, { ports }], i) => (
+        <div key={i}>
+          <li>{name}</li>
+          <ul> 
+            <li>
+              Ports:
+              <ul>
+                {Object.entries(ports).map(([port_name, port], i) => (
+                  <div key={i}>
+                    <li>
+                      <PartialValue 
+                        value={port_name} 
+                        errorPath={`protocols.${name}.ports.${port_name}`}
+                      />
+                    </li>
+                    <ul>
+                      {port.cidrs && (
+                        <li>
+                          IP addresses:
+                          <ul>
+                            {port.cidrs.map((address, i) => (
+                              <li key={i}>
+                                <PartialValue
+                                  value={address.CIDR}
+                                  errorPath={`protocols.${name}.ports.${port_name}.cidrs.${i}.CIDR`}
+                                />
+                              </li>
+                            ))}
```

Comment:
Ouff this is complex :'( Maybe split it in multiple small component but you can leave all the components in this file to avoid having 5 different files.